src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/References/ReferencesEndpoints.cs

```diff
@@ -0,0 +1,48 @@
+﻿using AutoMapper;
+using ECER.Utilities.Hosting;
+using MediatR;
+using Microsoft.AspNetCore.Http.HttpResults;
+
+namespace ECER.Clients.RegistryPortal.Server.References;
+
+public class ReferencesEndpoints : IRegisterEndpoints
+{
+  public void Register(IEndpointRouteBuilder endpointRouteBuilder)
+  {
+    endpointRouteBuilder.MapPost("/api/CharacterReference", async Task<Results<Ok, BadRequest<string>>> (CharacterReferenceSubmissionRequest request, IMediator messageBus, HttpContext httpContext, IMapper mapper, CancellationToken ct) =>
+    {
+      if (request.Token == null) return TypedResults.BadRequest("Token is required");
+      var result = await messageBus.Send(mapper.Map<Managers.Registry.Contract.Applications.CharacterReferenceSubmissionRequest>(request), ct);
+      if (!result.IsSuccess)
+      {
+        return TypedResults.BadRequest(result.ErrorMessage);
+      }
+      return TypedResults.Ok();
+    }).WithOpenApi("Handles character reference submission", string.Empty, "character_reference_post").WithParameterValidation();
+
+    endpointRouteBuilder.MapPost("/api/OptOutReference", async Task<Results<Ok, BadRequest<string>>> (OptOutReferenceRequest request, IMediator messageBus, HttpContext httpContext, IMapper mapper, CancellationToken ct) =>
```

Comment:
I'm not convinced this is how you'd want to structure the API, it is very RPC style right now. I also assume we have backlog stories to add recaptcha protections to these endpoint, right?

Reply:
Ya, We have stories to protect these endpoints using a captcha